tracer/src/Datadog.Trace.MSBuild/DatadogLogger.cs

```diff
@@ -106,7 +107,7 @@ private void EventSource_BuildStarted(object sender, BuildStartedEventArgs e)
 
                 _buildSpan = _tracer.StartSpan(BuildTags.BuildOperationName);
                 _buildSpan.SetMetric(Tags.Analytics, 1.0d);
-                _buildSpan.SetTraceSamplingPriority(SamplingPriorityValues.AutoKeep);
+                _buildSpan.Context.TraceContext?.SetSamplingPriority(SamplingPriorityValues.AutoKeep);
```

Comment:
My goal is to slowly move us towards a clear distinction between what is span-level metadata vs trace-level metadata. Here we are setting the sampling decision for the trace, so I think we should access the trace object (`TraceContext`) to change it. (Ideally we would access the trace directly from the span without going through the span context, like `span.Trace`. It's on my list of future refactorings.)

I also think it's a confusing API because it looks like you're setting sampling on a specific span, and then the tag could end up in a different span.

But I don't feel strongly about this (yet), so I can instead add an overload to the span extension that takes both sampling priority and sampling mechanism.